sysdig/resource_sysdig_secure_scanningpolicies.go

```diff
@@ -0,0 +1,249 @@
+			Read:   schema.DefaultTimeout(timeout),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"comment": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"isdefault": {
+				Type:     schema.TypeBool,
+				Computed: true,
+			},
+			"version": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "1_0",
+			},
+			"policy_bundle_id": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "default",
+			},
+			"rules": {
+				Type:     schema.TypeSet,
+				Required: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"id": {
+							Type:     schema.TypeString,
+							Computed: true,
+						},
+						"gate": {
+							Type:             schema.TypeString,
+							Required:         true,
+							ValidateDiagFunc: validateDiagFunc(validation.StringInSlice([]string{"always", "dockerfile", "files", "licenses", "metadata", "npms", "packages", "passwd_file", "retrieved_files", "vulnerabilities", "secret_scans", "ruby_gems"}, false)),
+						},
+						"trigger": {
+							Type:     schema.TypeString,
+							Required: true,
+							// ValidateDiagFunc: TODO: create inline func to validate each trigger options depending on gate https://developer.hashicorp.com/terraform/plugin/sdkv2/schemas/schema-behaviors
+						},
+						"action": {
+							Type:             schema.TypeString,
+							Required:         true,
+							ValidateDiagFunc: validateDiagFunc(validation.StringInSlice([]string{"WARN", "STOP"}, false)),
+						},
+						"params": {
+							Type:     schema.TypeSet,
+							Required: true,
+							// ValidateDiagFunc: TODO: function to validate name is valid for the given trigger,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"name": {
+										Type:     schema.TypeString,
+										Required: true,
+									},
+									"value": {
+										Type:     schema.TypeString,
+										Required: true,
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func resourceSysdigScanningPolicyCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client, err := meta.(SysdigClients).sysdigSecureClient()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	scanningPolicy := scanningPolicyFromResourceData(d)
+	scanningPolicy, err = client.CreateScanningPolicy(ctx, scanningPolicy)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	scanningPolicyToResourceData(&scanningPolicy, d)
+
+	return nil
+}
+
+func resourceSysdigScanningPolicyUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client, err := meta.(SysdigClients).sysdigSecureClient()
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	scanningPolicy := scanningPolicyFromResourceData(d)
+	id, _ := d.Get("id").(string)
```

Comment:
Shouldn't we get error here and check it? If its error return:
```
return diag.FromErr(err)
```
wdyt?

Reply:
uhm, that `d.Get("id")` is for the resource that previously has been created.
You run create, that creates in the state the resource with the ID and the run the update/read/delete. 
Checking further, don't think the `Get` returns error so the `_` should be removed. I took as a reference others so that's why is the same than other resources. 
Removing those for this PR.